### PR TITLE
Do not prefix remote_path IF WINDOWS

### PR DESCRIPTION
**Pull request by ruheenaansari34**
##### SUMMARY
https://github.com/ansible/ansible/pull/67832 fixes the need to run powershell modules on windows containers via docker connection. However, while running win_copy on windows containers, destination path is prefixed with / which doesn't work in the case of windows. The changes in this PR take care of bypassing that if OS is windows.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
Docker connection plugin (lib/ansible/plugins/connection/docker.py)

##### ADDITIONAL INFORMATION
For more context: https://github.com/felixfontein/ansible/pull/2#issuecomment-593591426

```paste below

```

### CODE REVIEW

**Comment by ruheenaansari34** [comment]
@felixfontein Opened this PR as per our conversation here https://github.com/felixfontein/ansible/pull/2#issuecomment-593591426

Thanks again, appreciate your active engagement :)

File: changelogs/fragments/67946-do_not_prefix_remote_path_if_windows
Comment (felixfontein) [inline changelogs/fragments/67946-do_not_prefix_remote_path_if_windows]: The filename needs to end with `.yml` or `.yaml` (that's what's causing CI to fail).

**Review by felixfontein — APPROVED** [review]
LGTM

I tested the plugin with Linux containers, and these still works fine. For Windows containers, this should definitively improve the situation.

Since there is no way to merge this to `devel` first anymore, I suggest this is directly merged to `stable-2.9`. Once the `community.general` repository is up and running, we'll also get this merged there.

**Comment by mattclay** [comment]
I understand the desire to merge this fix to stable-2.9 since devel is frozen. However, fixes should be merged into collections before being backported to the stable branches. I realize that's not possible at the moment since the transition isn't complete, but it should be possible in time for the next release.

I'm going to close this PR to help keep the backport queue clear. Once the fix has been merged in the collection, please open a new backport PR and reference the PR in the collection.

**Comment by felixfontein** [comment]
@ruheenaansari34 in that case I'm sorry, it won't be possible to get it into the next 2.9.x release. You have to wait until the collection repos are around. I will ping you here once that's possible and post instructions on how to do it (I also don't fully know yet :) ).

BTW, three's one thing you could already take a look at: there's also a function `fetch_file()` in the conncetion plugin which has the same problem (i.e. it also uses `_prefix_login_path()`). I think it is used when you use the [fetch](https://docs.ansible.com/ansible/latest/modules/fetch_module.html) module. Please note that it also does some more path arithmetic (it calls `os.path.dirname`, `os.path.join`, `os.path.basename`). It's probably best to move these calls into a `if` conditional and import `ntpath` (only in that `if` branch! use it instead of `os.path` -- see the notes in https://docs.python.org/3/library/os.path.html#module-os.path resp. https://docs.python.org/2/library/os.path.html#module-os.path). Maybe it also makes sense to keep calling `_prefix_login_path` and adjust that one to use `ntpath` as well for Windows containers. If you have any questions, feel free to ping me.

To make developing easier, you can also create a new PR in ansible/ansible for the `devel` branch; then I can comment in there. This PR will eventually be closed, but there will be a migration tool online eventually which should allow to easily migrate it to the collections repository. What do you think?

**Comment by ruheenaansari34** [comment]
Thanks @felixfontein for the details. It is unfortunate that this change is not going to be part of next release. 
Meanwhile, I will create a PR in `devel` with the suggestions you mentioned. But that would still mean, we would have to wait until collections is setup, merge this fix in collection and then create a backport PR for the stable release? Let me know if I understood this correctly :)

**Comment by felixfontein** [comment]
@ruheenaansari34 the fetch module docs explicitly say `This module is also supported for Windows targets.` so I'd assume it also works for Windows targets. I don't have any Windows target around to test, though :)

**Comment by ruheenaansari34** [comment]
Tested the changes against windows targets. They work well.

**Pull request closed without merging**
